### ISSUE TITLE
Errors in 'awsrecipes_rotate_my_key.py' when trying to rotate keys with multiple profiles.

### ISSUE THREAD

**Issue by grettir** [nccgroup/AWS-recipes#19]
This is probably an issue with the underlying opinel library, but I've run into a bizarre instance where if I have two profiles in my `~/.aws/credentials` file, the `awsrecipes_rotate_my_key.py` script will work if I'm trying to rotate keys for the ***first*** profile in the list, but will fail for the ***second***.

For instance, if my `~/.aws/credentials` file looks something like this:

```
[test1]
*access key and token info*

[test2]
*access key and token info*

[test1-nomfa]
*access key info*

[test2-nomfa]
*access key info*
```

Rotating keys for the `test1` profile works as expected:

```
» ./awsrecipes_rotate_my_key.py --profile test1
Connecting to AWS iam...
Fetching username...
Creating a new access key for 'xxxxxxx'...
User 'xxxxxxx' currently has 2 access keys:
	XXXXXXXXXXXXXXXXXXXX (Active)
	XXXXXXXXXXXXXXXXXXXX (Active)
Initiating a session with the new access key...
Enter your MFA code (or 'q' to abort):
XXXXXX
Connecting to AWS iam...
Deleting the old access key...
User 'xxxxxxx' currently has 1 access keys:
	XXXXXXXXXXXXXXXXXXXX (Active)
Success !
```

But, if I try to rotate the keys for the second profile, `test2`, it results in the following:

```
» ./awsrecipes_rotate_my_key.py --profile test2
Connecting to AWS iam...
Fetching username...
Creating a new access key for 'xxxxxxx'...
User 'xxxxxxx' currently has 2 access keys:
	XXXXXXXXXXXXXXXXXXXX (Active)
	XXXXXXXXXXXXXXXXXXXX (Active)
Traceback (most recent call last):
  File "./awsrecipes_rotate_my_key.py", line 106, in <module>
    sys.exit(main())
  File "./awsrecipes_rotate_my_key.py", line 71, in main
    write_creds_to_aws_credentials_file(profile_name + '-nomfa', new_credentials)
  File "/usr/local/lib/python2.7/site-packages/opinel/utils/credentials.py", line 356, in write_creds_to_aws_credentials_file
    complete_profile(f, credentials['SessionToken'], session_token_written, credentials['SessionToken'], mfa_serial_written)
KeyError: 'SessionToken'
```

The new key for the `test2` profile ***is*** created and written to the `~/.aws/credentials` file, but the new keys *isn't* validated and the old key *isn't* deleted (leaving me with two active keys).

My first though was that there must be something wrong with that second `test2` profile, but just out of curiosity, I swapped the order of the `test1` and `test2` profiles like so:

```
[test2]
*access key and token info*

[test1]
*access key and token info*

[test2-nomfa]
*access key info*

[test1-nomfa]
*access key info*
```

…and I got the exact opposite results:

  - Rotating keys for the `test2` profile worked as expected.
  - Rotating keys for the `test1` profile failed.

I've been able to reproduce this multiple times, so it appears that the failure really is dependent on the order of the profiles in my `~/.aws/credentials` file.

Is anyone else seeing this behavior?

**Comment by l01cd3v** [comment]
I was able to repro this issue, though I am pretty sure it is not an opinel issue because this happened to both profiles in my case. Instead I believe this was a timing issue w/ the key being used too soon after creation. In the non-STS case, there was a 5 second sleep.

I pushed a code change that forces the sleep in both cases (when using MFA and not), and it seems to do the trick. Let me know if you're still experiencing this issue with the new code
